plugins/aws/s3/bucketEncryptionInTransit.js

```diff
@@ -86,5 +97,98 @@ module.exports = {
             }
         }
         callback(null, results, source);
+    },
+    remediate: function(config, cache, settings, resource, callback) {
+        var putCall = this.actions.remediate;
+        var pluginName = 'bucketEncryptionInTransit';
+        var bucketNameArr = resource.split(':');
+        var bucketName = bucketNameArr[bucketNameArr.length - 1];
+
+        // find the location of the bucket needing to be remediated
+        var bucketLocations = cache['s3']['getBucketPolicy'];
+        var bucketLocation;
+
+        for (var key in bucketLocations) {
+            if (bucketLocations[key][bucketName]) {
+                bucketLocation = key;
+                var policy = bucketLocations[key][bucketName];
+                break;
+            }
+        }
+
+        // add the location of the bucket to the config
+        config.region = bucketLocation;
+
+        // create the params necessary for the remediation
+        var params = {};
+        var SecureTransport = {
+            'Sid':'DenyInSecureTransport',
+            'Effect': 'Deny',
+            'Principal': '*',
+            'Action': 's3:*',
+            'Resource': [resource,resource+'/*'],
+            'Condition': {
+                'Bool': {
+                    'aws:SecureTransport': 'false'
+                }
+            }
+        };
+        var policyBody = {
+            'Version': '2012-10-17'
+        };
+        if (policy.err &&
+            policy.err.code === 'NoSuchBucketPolicy'){
+            policyBody['Statement'] = SecureTransport;
+            params = {
+                'Bucket': bucketName,
+                'Policy': JSON.stringify(policyBody)
+            };
+        } else {
+            var policyJson;
+            if (typeof policy.data.Policy === 'string') {
```

Comment:
check if policy.data and policy.data.Policy exist before doing comparison

```suggestion
            if (policy.data && policy.data.Policy && typeof policy.data.Policy === 'string') {
```

Reply:
done